Runtime/CSharp/Bitcraft.StateMachine/StateManager.cs

```diff
@@ -75,26 +75,28 @@ public void SetInitialState(StateToken initialState, object data)
             RaiseOnExitEvent(null, null);
 
             CurrentState = null;
-            PerformTransitionTo(initialState, data);
```

Comment:
The problem here and on completion is that it's not possible to make the difference between the two.
I was thinking of having special actions defined in the `StateManager` itself, such as `InitializeStateMachineAction` and `FinalizeStateMachineAction`. So then the user can still compare the action it receives against something to make sure.

Let's discuss about it.

Reply:
Just message for myself, after trying to make sense of the previous sentence after more than two years, the "two" in "difference between the two" refers to

1. passing `null` (ActionToken) when the state machine goes to initial state (there is no user action to trigger that, it's internal operation, there is not even a source state in the transition info).
2. passing `null` (StateToken) when requesting the state machine to go to final state.

Then, I just realize all this make no sense, as they are two different things. Let's ignore this for the moment as it may not even be a problem. Let's revisit all this after gaining more experience based on usage.